src/Finch_Run.hpp

```diff
@@ -0,0 +1,121 @@
+/****************************************************************************
+ * Copyright (c) 2024 by Oak Ridge National Laboratory                      *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of Finch. Finch is distributed under a                 *
+ * BSD 3-clause license. For the licensing terms see the LICENSE file in    *
+ * the top-level directory.                                                 *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+#ifndef SingleLayer_H
+#define SingleLayer_H
+
+#include <Cabana_Grid.hpp>
+#include <Kokkos_Core.hpp>
+
+#include "Finch_Grid.hpp"
+#include "Finch_Inputs.hpp"
+#include "Finch_SolidificationData.hpp"
+#include "Finch_Solver.hpp"
+#include "MovingBeam/Finch_MovingBeam.hpp"
+
+namespace Finch
+{
+
+template <typename MemorySpace>
+class SingleLayer
```

Comment:
I don't know if there are any plans to make finch into a code that simulates multiple layers with heat accumulation from prior simulated layers (@colemanjs can correct me if I'm wrong here), but the name `SingleLayer` is a little confusing because it implies that we're simulating/can simulate multiple layers when we aren't. Even with future in-memory coupling using ExaCA, we'd only simulate one layer at a time, I feel like something like `Layer` might be a better name? This is probably just me being picky though

Reply:
I agree - fixed